### DISCUSSION TITLE
Chnage FXGL type from class to interface?

### DISCUSSION THREAD

**Discussion by chengenzhao** [AlmasB/FXGL#1192] [Q&A]
Hi:

Currently we use Java to write the game
but one problem is we are unable to extend FXGL class
and it won't be a problem if we use Kotlin as long as we use extension 
but it is a problem in Java world, we could only use extends to expand the type while FXGL has private constructor so it doesn't allow us to extends the class
so is it possible to change FXGL type from class to interface?
<img width="740" alt="截屏2022-07-31 下午5 58 33" src="https://user-images.githubusercontent.com/5525436/182021005-f600877d-26f1-44f8-92da-87a9bee04153.png">
All methods in the FXGL are @JVMStatic, and Java supports static method in the interface now
so I think it should be fine to use interface instead of class with private constructor

Thanks

**Comment by AlmasB** [comment]
Hi, the only use case I can think of for this is to get all methods into the scope of the class extending FXGL, but the same can be achieved with `import static ...`, so I don't think this change is justified.
